src/main/java/org/spongepowered/asm/mixin/injection/struct/MemberInfo.java

```diff
@@ -443,13 +443,6 @@ public MemberInfo validate() throws InvalidMemberDescriptorException {
             if (!this.owner.matches("(?i)^[\\w\\p{Sc}/]+$")) {
                 throw new InvalidMemberDescriptorException("Invalid owner: " + this.owner);
             }
-            try {
-                if (!this.owner.equals(Type.getType(this.owner).getDescriptor())) {
```

Comment:
This line (attempts) to convert `owner` from a String to a Type and then back into a String to make sure that it is a valid Type. The problem is that `owner` is stored without `L;` (which is used in the JVM to notate a Class), while [`getType`](https://github.com/llbit/ow2-asm/blob/a695934043d6f8f0aee3f6867c8dd167afd4aed8/src/org/objectweb/asm/Type.java#L479-L484) and [`getDescriptor`](https://github.com/llbit/ow2-asm/blob/a695934043d6f8f0aee3f6867c8dd167afd4aed8/src/org/objectweb/asm/Type.java#L668-L670) use `L;`. What we want to do here is create a copy of `owner` that contains the `L;` and test against that.

```java
String ownerType = "L" + this.owner + ";";
if (!ownerType.equals(Type.getType(ownerType).getDescriptor())) {
```